src/Disks/ObjectStorages/AzureBlobStorage/AzureObjectStorage.cpp

```diff
@@ -193,8 +193,7 @@ void AzureObjectStorage::listObjects(const std::string & path, RelativePathsWith
                 break;
             options.PageSizeHint = keys_left;
         }
-
-        if (blob_list_response.HasPage())
+        if (blob_list_response.NextPageToken.HasValue() && !blob_list_response.NextPageToken.Value().empty())
```

Comment:
I've found this way to check in example https://github.com/Azure/azure-sdk-for-cpp/blob/azure-storage-blobs_12.10.0/sdk/storage/azure-storage-blobs/samples/blob_list_operation.cpp#L53-L71. Is it correct one?

Reply:
I think it is correct under the condition that `MoveToNextPage()` is used as an iterator:
``` cpp
for (auto containerPage = serviceClient.ListBlobContainers(); containerPage.HasPage();
       containerPage.MoveToNextPage())
```

It doesn't work with `options.ContinuationToken = blob_list_response.NextPageToken;`

Reply:
Checking with HasPage(), MoveToNextPage()

Reply:
So the issue is that `BlobContainerClient::ListBlobs` sets a page, thus the sequential `blob.HasPage()` condition is `true`:
```
2024.05.03 01:43:16.438300 [ 8 ] {} <Trace> AzureObjectStorage before ListBlobs: HasPage: false
2024.05.03 01:43:16.440368 [ 8 ] {} <Trace> AzureObjectStorage after ListBlobs: HasPage: true
```
And the loop never breaks...

I modified the fix so it follows the example. Namely, the condition `HasPage()` is verified before listing blobs.